KinomeScan-Appyter/Kinase Input Data Processing

```diff
@@ -0,0 +1,69 @@
+# Creates the dictionaries and converts them to GMT files
+
+import pandas as pd
+
+old_url = 'http://lincs.hms.harvard.edu/db/datasets/20000/results?small+molecules=HMS_ID&output_type=.csv'
+
+df = []
+
+kinome_df = pd.read_excel('http://lincs.hms.harvard.edu/wordpress/wp-content/uploads/2013/11/HMS-LINCS_KinomeScan_Datasets_2018-01-18.xlsx')
+kinome_df['sm_hms_id'] = kinome_df['sm_hms_id'].str.replace(r'\D', '') #remove HMSL before the ID
+
+# retrieve the CSV file for the small molecules, 2-183
+for x in range(0, 181):  
+    hms_id_string = str(kinome_df.iloc[x]['sm_hms_id'])
+    new_url = old_url.replace("HMS_ID", hms_id_string) # fills in HMS_ID with the correct small molecule 
+    
+    data = pd.read_csv(new_url)
+    df.append(pd.DataFrame(data))
+
+# create dictionaries, one corresponding to each target affinity level 
+dict_1 = {}
+dict_2 = {}
+dict_3 = {} 
+
+# kinases are keys and small molecules are values 
+for sm in df:
+    for x in range(0, sm.shape[0] - 1):
+        kinase_name = sm.iloc[x]['HUGO Gene Symbol']
+        sm_name = sm.iloc[0]['HMSL Small Mol Name']
+        target_aff = sm.iloc[x]['Binding Class']
+        
+        if (target_aff == 1):
+            if (kinase_name in dict_1):
```

Comment:
There's a better and cleaner way to do it - initialize levels as `defaultdict(list)`. This way you can do `level_1.append(sm_name)` without any condition checks

Reply:
So defaultdict automatically checks for duplicates?

Reply:
If you want to collapse duplicates, you can use `defaultdict(set)`